core/src/Streamly/Internal/Data/Parser.hs

```diff
@@ -1199,13 +1201,15 @@ takeP i p = D.toParserK $ D.takeP i $ D.fromParserK p
 --
 -- This is same as 'Data.Traversable.sequence' but more efficient.
 --
--- /Unimplemented/
+-- /Pre-release/
 --
 {-# INLINE concatSequence #-}
 concatSequence ::
-    -- Foldable t =>
+    (MonadCatch m, Foldable t) =>
     Fold m b c -> t (Parser m a b) -> Parser m a c
```

Comment:
Instead of using a Foldable t, you can use `SerialT m (Parser m a b)` here, it would be more general as we can easily convert a Foldable to stream.